pkg/protocol/engine/ledger/ledger/ledger.go

```diff
@@ -74,8 +76,10 @@ func NewProvider() module.Provider[*engine.Engine, ledger.Ledger] {
 			// TODO: how do we want to handle changing API here?
 			iotagoAPI := l.apiProvider.CurrentAPI()
 			l.manaManager = mana.NewManager(iotagoAPI.ManaDecayProvider(), iotagoAPI.ProtocolParameters().RentStructure(), l.resolveAccountOutput)
```

Comment:
the manaManager should probably also take the provider directly and not the current api